pkg/engine/template_generator.go

```diff
@@ -205,6 +206,29 @@ func (t *TemplateGenerator) getMasterCustomData(cs *api.ContainerService, textFi
 // getTemplateFuncMap returns all functions used in template generation
 func (t *TemplateGenerator) getTemplateFuncMap(cs *api.ContainerService) template.FuncMap {
 	return template.FuncMap{
+		"IsAzureStackCloud": func() bool {
+			var cloudProfileName string
+			if cs.Properties.CustomCloudProfile != nil {
+				if cs.Properties.CustomCloudProfile.Enviornment != nil {
+					cloudProfileName = cs.Properties.CustomCloudProfile.Enviornment.Name
+				}
+			}
+			return strings.EqualFold(cloudProfileName, api.AzureStackCloud)
+		},
+		"GetCustomEnvironmentJSON": func() string {
+			var environmentJSON string
+			if cs.Properties.CustomCloudProfile != nil {
+				if cs.Properties.CustomCloudProfile.Enviornment != nil {
+					bytes, err := json.Marshal(cs.Properties.CustomCloudProfile.Enviornment)
```

Comment:
@tariq1890 should we move this "get JSON representation" logic to be a method of the `Environment` type struct?

Reply:
GetCustomEnvironmentJSON() will return empty string if CustomCloudProfile field is empty.
So I put the method to struct Properties, also move IsAzureStackCloud() to struct Properties.